lib/minitest/stub_const.rb

```diff
@@ -37,15 +36,21 @@ def stub_const(name, val, &block)
   # Example:
   #
   #   Object.stub_remove_const(:File) do
-  #     "Look ma, no File!" unless defined(File)
+  #     "Look ma, no File!" unless defined?(File)
   #   end
   #   # => "Look ma, no File!"
   def stub_remove_const(name)
-    orig = const_get(name)
-    remove_const(name)
-    yield
-  ensure
-    const_set(name, orig)
+    if const_defined?(name)
+      begin
+        orig = const_get(name)
+        remove_const(name)
+        yield
+      ensure
+        const_set(name, orig)
+      end
+    else
+      yield
```

Comment:
Given that this method is named `stub_remove_const` rather than `ensure_const_stubbed`, I think that we should raise if the given const is undefined, rather than silently yield. Do you have a specific use-case for this in mind?